tsdb/errors/errors.go

```diff
@@ -79,7 +79,31 @@ func (es nonNilMultiError) Error() string {
 	return buf.String()
 }
 
-// CloseAll closes all given closers while recording error in MultiError.
+// As finds the first error in multiError slice of error chains matches target, and if so, sets
+// target to that error value and returns true. Otherwise, it returns false.
+func (es multiError) As(target interface{}) bool {
+	for _, err := range es.errs {
+		if stderrors.As(err, target) {
+			return true
+		}
+	}
+	return false
+}
+
+// Is reports whether any error in multiError's slice of error chains matches target.
+func (es multiError) Is(target error) bool {
+	if _, ok := target.(multiError); ok {
+		return true
+	}
```

Comment:
Should we also require all errors in `target` to match with `es`? Or is it just the _type_ of the error that should match?

Reply:
As written in docs.. it's up to us:

```
// Is reports whether any error in err's chain matches target.
//
// The chain consists of err itself followed by the sequence of errors obtained by
// repeatedly calling Unwrap.
//
// An error is considered to match a target if it is equal to that target or if
// it implements a method Is(error) bool such that Is(target) returns true.
//
// An error type might provide an Is method so it can be treated as equivalent
// to an existing error. For example, if MyError defines
//
//	func (m MyError) Is(target error) bool { return target == os.ErrExist }
//
// then Is(MyError{}, os.ErrExist) returns true. See syscall.Errno.Is for
// an example in the standard library.
```

Will add comment (:

Reply:
You are right.

< Offline 1:1 with @codesome >